console/docable.ts

```diff
@@ -0,0 +1,91 @@
+const decoder = new TextDecoder();
+const encoder = new TextEncoder();
+
+/**
+ * Parse data member doc blocks and signatures and place them in a minimalistic,
+ * JSON format.
+ */
+export class Docable {
+
+  protected filepaths: string[];
+  protected json_output: {[k: string]: string[]} = {};
+  protected output_filepath: string;
+
+  constructor(filepaths: string[], outputFilepath: string) {
+    this.filepaths = filepaths;
+    this.output_filepath = outputFilepath;
+  }
+
+  public async run() {
+    for (let index in this.filepaths) {
+      const filepath = this.filepaths[index];
+
+      const fileContents = await this.getFileContents(filepath);
+
+      // Get the full member name and only continue with the script if it's found
+      const fullMemberName = this.getFullMemberName(fileContents);
+      if (!fullMemberName) {
+        console.log(`File "${filepath}" is missing the "/// Member:" comment at the top of the file.`);
+        Deno.exit(1);
+      }
+
+      this.json_output[fullMemberName as string] = [];
```

Comment:
nitpick again.. dont suppose this could be moved to be above the foreaach loop below? just because it's not used until that block

Reply:
3604b6c